### PR TITLE
Update the date_identifier parser rule to accept single quotes in date

### PR DESCRIPTION
**Pull request by dsantos**
Tidy up the calendar-query build
Update the date_identifier parser rule to accept single quotes in date

### CODE REVIEW

**Review by mattvickery — COMMENTED** [review]
Was there any particular reason to remove the compiler plugin froths module?

**Review by mattvickery — CHANGES_REQUESTED** [review]
I am happy for the commons-lang3 dependency to be removed rather than commented out, it was lazy (on my part).

**Review by mattvickery — CHANGES_REQUESTED** [review]
There is no need to use Optional.ofNullable(...) because the Lexer error listener will be used to trigger failure on missing tokens. By the time we get to parsing, we can guarantee that the query is well formed, i.e. lexically correct. I will add the lever error listener and activate it in the next few days.

**Comment by mattvickery** [comment]
Thanks Duarte, some good and really useful changes in there.

File: query/pom.xml [hunk 1]
```diff
@@ -52,39 +46,6 @@
 
     <build>
         <plugins>
-            <plugin>
```
Comment (dsantos) [inline query/pom.xml:55]: The compiler plugin will be inherited from the parent pom.

File: query/pom.xml [hunk 2]
```diff
@@ -26,22 +26,11 @@
             <artifactId>antlr4</artifactId>
             <version>${antlr.version}</version>
         </dependency>
-        <!--<dependency>-->
```
Comment (duartesantos) [inline query/pom.xml]: Removed

**Review by dsantos — COMMENTED** [review]
All changes addressed. Thanks

File: query/pom.xml (hunk 1 above)
Comment (mattvickery) [reply query/pom.xml:55]: Ah, well spotted.

**Review by mattvickery — APPROVED** [review]

**Pull request merged**